virtio-devices/src/balloon.rs

```diff
@@ -351,6 +358,10 @@ impl Balloon {
     pub fn resize(&self, size: u64) -> Result<(), Error> {
         self.resize.work(size)
     }
+
+    pub fn get_actual(&self) -> u64 {
+        (self.config.lock().unwrap().actual as u64) << PAGE_SHIFT
```

Comment:
There's a problem with the size of `PAGE_SHIFT` here. `PAGE_SHIFT` is always `4kiB` but I think that's incorrect because based on the way the memory was mapped, hugepages might be in use, and that means the `PAGE_SHIFT` should be `2MiB` instead.

Reply:
I think this part should use VIRTIO_BALLOON_PFN_SHIFT like qemu.

Reply:
Ok I read the `virtio-balloon` driver in Linux and you're right:
```
/*
 * Balloon device works in 4K page units.  So each page is pointed to by
 * multiple balloon pages.  All memory counters in this driver are in balloon
 * page units.
 */
#define VIRTIO_BALLOON_PAGES_PER_PAGE (unsigned)(PAGE_SIZE >> VIRTIO_BALLOON_PFN_SHIFT)
```
It does not matter what kind of pages the guest is using, 4k is always the unit used for `virtio-balloon`.